cdap-ui/app/cdap/components/DataPrep/Directives/Hash/index.tsx

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { Component } from 'react';
+import classnames from 'classnames';
+import { execute } from 'components/DataPrep/store/DataPrepActionCreator';
+import T from 'i18n-react';
+import DataPrepStore from 'components/DataPrep/store';
+import DataPrepActions from 'components/DataPrep/store/DataPrepActions';
+import MouseTrap from 'mousetrap';
+import { setPopoverOffset } from 'components/DataPrep/helper';
+import { preventPropagation } from 'services/helpers';
+
+require('./Hash.scss');
+
+const PREFIX = 'features.DataPrep.Directives.Hash';
+
+interface IHashProps {
+  column: string;
+  onComplete: () => void;
+  isOpen: boolean;
+  close: () => void;
+}
+
+export default class Hash extends Component<IHashProps> {
```

Comment:
@sunxd-chromium One last comment. This directive applies only to string and byte[]. I remember us talking about including that as a condition to enable/disable this directive in the dropdown. It can be done similar [to this](https://github.com/cdapio/cdap/blob/develop/cdap-ui/app/cdap/components/DataPrep/Directives/Format/index.js#L58).